02-express-tutorial/app.js

```diff
@@ -1,77 +1,77 @@
-} 
-return res.status(200).json(product)
-})
-//exercise 12
-app.get("/api/v1/query", (req, res)=> {
-    console.log(req.query)
-    const {search, limit,priceLessThan} = req.query
-    let sortedItems = [...products]
-    if(search){
-        sortedItems=sortedItems.filter((product)=> {
-        return product.name.startsWith(search)
-        })
-    }
-    if(limit){
-         sortedItems=sortedItems.slice(0, Number(limit))
-    } 
-    //exercise 13
-    if(priceLessThan){
-         sortedItems=sortedItems.filter((product)=>{
-        return product.price<priceLessThan
-        })
-    }
-    // if(sortedItems.length<1){
-    //     // res.status(200).send("no products match your search")
-    //    return  res.status(200).json({success: true, msg: "no items founded"})
-    // }
-    return  res.status(200).json({success: true, data: sortedItems})
-//   return  res.status(200).json(sortedItems)
-    // res.send("hello world")
+//create a middleware function logger
+const logger = (req, res, next) => {
+  const method = req.method;
+  const url = req.url;
+  const time = new Date().toTimeString();
+  console.log(method, url, time);
+  next();
+};
 
-})
+const auth = (req, res, next) => {
+  if (!req.cookies.name) {
+    res.status(401).json({ success: false, msg: "Unauthorized" });
+  } else {
+    req.user = req.cookies.name;
+    next();
+  }
+};
 
+app.use(express.urlencoded({ extended: false }));
+app.use(express.json());
 
-//exercise 4
-// app.post()
-//exercise 6
-app.all("*", (req,res)=> {
-    res.status(404).send("Page not found")
-})
+//" take the logger call out of your app.get() statement, and call it via app.use(), for all paths, instead. Verify that it still works."
+app.use(logger);
+//" Change the directory for your static serving from ./public to ./methods-public, and try out that frontend from your browser."
 
-app.listen(3000, ()=> {
-console.log('Express Tutorial')
-})
+app.use(express.static("./methods-public"));
+// app.use(express.static("./public"))
 
+app.use("/api/v1/people", peopleRouter);
+app.use("/api/v1/products", productsRouter);
 
-// The require statement to import the express module
-// Creation of the app as returned from calling express()
-// app.use statements for the middleware. You’ll eventually use many kinds of middleware, but for now the only middleware we are using is express.static().
-// app.get and app.post statements for the routes you will handle. Eventually these will be refactored into router modules, but for now you can put them inline.
-// An app.all statement after these to handle page not found conditions.
-// An app.listen statement to tell the server to listen for requests on a particular port.
\ No newline at end of file
+//"Call your logger using the first method, in one of your app.get() statements, and verify that it works. "
+// app.get("/api/v1/test", logger, (req, res)=> {
+
+app.post("/login", (req, res) => {
```

Comment:
This method is the same as the logon method, right? I think you can remove one of them.